tanjun/dependencies/data.py

```diff
@@ -149,7 +149,8 @@ def make_lc_resolver(type_: type[_T], /) -> collections.Callable[..., collection
     """
 
     async def resolve(
-        constant: LazyConstant[_T] = injecting.inject(type=LazyConstant[type_]),
+        # LazyConstant gets type arguments at runtime
+        constant: LazyConstant[_T] = injecting.inject(type=LazyConstant[type_]),  # type: ignore[valid-type]
```

Comment:
`type_` is a runtime variable, mypy doesn't let runtime things like that be a type arg (the things in brackets)

This could probably be put behind an `if MYPY` though.